provider/blob/src/blob_schema.rs

```diff
@@ -160,7 +169,12 @@ impl<'data, LocaleVecFormat: VarZeroVecFormat> BlobSchemaV3<'data, LocaleVecForm
             .buffers
             .get(blob_index)
             .ok_or_else(|| DataError::custom("Invalid blob bytes").with_req(marker, req))?;
-        Ok(buffer)
+        Ok((
+            buffer,
+            ZeroTrieSimpleAscii::from_store(zerotrie)
+                .get(CHECKSUM_KEY)
+                .map(|cs| cs as u64),
```

Comment:
Issue/Discussion: Not super happy that we run this all the time on every request even though most keys don't have a checksum.

Reply:
Well it's still a binary search.

If you want to get the very first item, you can use the constant-time `probe` function instead.

Reply:
Also, due to varint packing inefficiencies for large values, this probably ends up taking 10 bytes.